Sources/WebSocketKit/WebSocket.swift

```diff
@@ -42,6 +49,14 @@ public final class WebSocket {
     public func onBinary(_ callback: @escaping (WebSocket, ByteBuffer) -> ()) {
         self.onBinaryCallback = callback
     }
+    
+    public func onPong(_ callback: @escaping (WebSocket) -> ()) {
```

Comment:
Can both client and server send a ping or is it always client sending ping and server pong? 

I wonder because it might be confusing which one to override depending on whether your WebSocket is a client or server one.

I also think it could potentially be confusing as to whether _you_ need to do anything on this callback. For example, it seems like in `onPing` you might need to write back a `pong`. What if that's what you are trying to do be overriding this? 

Maybe it would be clearer to name this something like `didPong` or something. 

Thoughts?

Reply:
It's really just an opcode on WebSockets. Either Server or Client can send a ping (knowing a pong has come back can be useful), under tricky network conditions a ping/pong receipt is really useful for us when debugging/profiling for example.

I wonder on a server whether more information could be needed to make this a meaningful callback.

I agree it could be slightly confusing. Happy to change it to `didPong`, only reason it was `onPong` was because of `onPing`.

Reply:
I'd say the same thing for `onPing`, too. In both `onText` and `onBinary` you are expected to handle the incoming message and do something. For `onPing`, we're essentially just letting you know it happened since the WebSocket library handles sending the pong automatically. 

We could name it `onPing` and just document this but I wonder if we could do better. That's why I was thinking something like:
- `didPing` / `didPong`
- `didReceivePing` / `didReceivePong`
- `onSendPong`, `onReceivePong`

Another option is we call it `onPing` and expect the user to do `ws.sendPong()`. That could be the default implementation of the handler. Maybe that's the best option?

Reply:
I think `didReceivePing` and `didReceivePong` is probably the best approach. Would this be a breaking change?

I would say the library should respond with a pong, don't normally like to retort with 'it's in the spec', but it is! 😄 `Upon receipt of a Ping frame, an endpoint MUST send a Pong frame`.

In an unrelated note: It's also interesting that sending pings to API Gateway websockets certainly does not keep the connection alive!